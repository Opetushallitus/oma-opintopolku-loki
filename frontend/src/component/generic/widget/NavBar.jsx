import React from 'react'
import styled from 'styled-components'
import constants from 'ui/constants'
import media from 'ui/media'
import t from 'util/translate'

const Navigation = styled.div`
  display: flex;
  margin-bottom: 1.714rem;
`

const Tab = styled.span`
  display: inline-block;
  padding: 0.5rem 2rem;
  
  ${media.full`
    padding: 0.5rem 5rem;
  `}
`

const InactiveTab = styled(Tab)`
  box-sizing: border-box;
  height: 100%;
  font-weight: 600;
  background-color: ${constants.color.background.neutralLight};
  border-bottom: 1px solid ${constants.color.primary};
`

const ActiveTab = styled(Tab)`
  border-color: ${constants.color.primary} ${constants.color.border} ${constants.color.white} ${constants.color.border};
  border-style: solid solid none solid;
  border-width: 2px 1px 0px 1px;
`

const Filler = styled.span`
  flex: 1;
  border-bottom: 1px solid ${constants.color.primary};
`

const Link = styled.a`
  color: ${constants.color.black};
`

const NavBar = () => (
  <Navigation>
    <Link href={'/koski/omadata/kayttooikeudet'}>
      <InactiveTab>{t`Annetut käyttöluvat`}</InactiveTab>
    </Link>
    <ActiveTab>{t`Tietojani käyttäneet toimijat`}</ActiveTab>
    <Filler />
  </Navigation>
)

export default NavBar
