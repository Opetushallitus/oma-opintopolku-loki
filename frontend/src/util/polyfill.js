import 'core-js/features/promise'
import 'core-js/features/string/starts-with'
