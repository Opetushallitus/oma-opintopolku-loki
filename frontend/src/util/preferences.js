import Cookies from 'js-cookie'

export const lang = Cookies.get('lang') || 'fi'
