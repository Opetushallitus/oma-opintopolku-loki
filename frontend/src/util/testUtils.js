/* istanbul ignore file */
import renderer from 'react-test-renderer'

export function matchesSnapshot (component) {
  const renderedComponent = renderer.create(component)
  const componentJSON = renderedComponent.toJSON()
  expect(componentJSON).toMatchSnapshot()
  renderedComponent.unmount()
}
