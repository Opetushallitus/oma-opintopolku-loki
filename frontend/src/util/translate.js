import Cookies from 'js-cookie'

export default key => {
  const lang = Cookies.get('lang') || 'fi'
  const translations = window.translationsMap || {}
  const versions = translations[key] || {}
  const translated = versions[lang]

  if (!translated) {
    console.error(`Translation missing for language ${lang}: ${key}`)
    return key
  }

  return translated
}
