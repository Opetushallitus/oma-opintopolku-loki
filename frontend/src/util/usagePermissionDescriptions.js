import t from 'util/translate'

import permissionsMap from 'Resources/mapping/usagePermissionDescriptions'

export const getTranslatedUsagePermissionDescription = organizationOid => {
  const permission = permissionsMap[organizationOid] || permissionsMap.default
  return t(permission)
}

export const isMydataPartner = organizationOid => {
  return Boolean(permissionsMap[organizationOid])
}
