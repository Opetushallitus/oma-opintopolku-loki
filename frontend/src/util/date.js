import moment from 'moment'
import { lang } from 'util/preferences'

moment.locale(lang)

export const isoStringToDate = isoString => moment(isoString).format('l, LT')
