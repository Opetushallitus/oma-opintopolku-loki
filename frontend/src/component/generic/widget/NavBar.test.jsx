import React from 'react'
import NavBar from './NavBar'
import { matchesSnapshot } from 'util/testUtils'

describe('NavBar', () => {
  it('should be rendered', () => {
    matchesSnapshot(<NavBar/>)
  })
})
