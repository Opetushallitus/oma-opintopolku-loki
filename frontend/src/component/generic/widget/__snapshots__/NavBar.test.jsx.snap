// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should be rendered 1`] = `
<div
  className="NavBar__Navigation-sc-18mzj0n-0 jHKwkt"
>
  <a
    className="NavBar__Link-sc-18mzj0n-5 ktidaL"
    href="/koski/omadata/kayttooikeudet"
  >
    <span
      className="NavBar__Tab-sc-18mzj0n-1 NavBar__InactiveTab-sc-18mzj0n-2 dhKxFl"
    >
      Annetut käyttöluvat
    </span>
  </a>
  <span
    className="NavBar__Tab-sc-18mzj0n-1 NavBar__ActiveTab-sc-18mzj0n-3 erWAMW"
  >
    Tietojani käyttäneet toimijat
  </span>
  <span
    className="NavBar__Filler-sc-18mzj0n-4 jTKRzr"
  />
</div>
`;
